service/storage_proxy.cc

```diff
@@ -2141,16 +2141,24 @@ gms::inet_address storage_proxy::find_leader_for_counter_update(const mutation&
         throw exceptions::unavailable_exception(cl, block_for(ks, cl), 0);
     }
 
+    const auto my_address = utils::fb_utilities::get_broadcast_address();
+    for (const auto& ep : live_endpoints) {
+        // Favour ourselves to avoid additional hop
```

Comment:
You do no need to search in `live_endpoints`, but in `local_endpoint` and you can do it with `boost::find`.

Reply:
Agreed with `boost::find`, but I think that (with modern drivers) this early return will fire almost always. So by exiting before the creation of `local_endpoints` we save a vector copy at the cost of iterating longer table.

Anyway, it's not worth that much discussion. If you are not convinced with the above please just say "no".

Reply:
See my other comment here: https://github.com/scylladb/scylla/pull/8282#discussion_r595048611